### DISCUSSION TITLE
Add a class to a DIV tag using traits

### DISCUSSION THREAD

**Discussion by MarcoZucc** [GrapesJS/grapesjs#5200] [Q&A]
I can't solve a problem related to traits in grapejs

I created a traits that acts on the DIV and SECTION tags

It should give the possibility to add a pre-determined class (contained in a drop-down) to a DIV element.
On data attributes it works for me.
If I add a class to a DIV that contains another class (example myclass) it does not work: I delete the old class. 
Instead, it should add the new class to the old one. 

Try adding the block "SECTION" to the editor: it has its own class "myclass" that is deleted if through the “component setting” I go to edit it 

Has anyone already done something similar? I would then implement everything to use tailwind and its classes
I have made a working demo

[https://jsfiddle.net/zucco/5jnt9zhL/17/](https://jsfiddle.net/zucco/5jnt9zhL/17/)

**Comment by artf** [comment]
The component updates automatically its classes if you update `attributes.class` (this is what your trait does), so you can simply choice another attribute name for storing your color classes and then update the real classes container on change.

**Reply by MarcoZucc** [reply]
basic behavior of the traits is as you described it: artf you are right.
But I would like to take the original classes and add a class using the traits. 
Unfortunately the traits overrides all the classes that were originally associated with the block
